internal/unifi/client.go

```diff
@@ -80,56 +98,109 @@ func (c *httpClient) login() error {
 	// Check if the login was successful
 	if resp.StatusCode != http.StatusOK {
 		respBody, _ := io.ReadAll(resp.Body)
-		log.Error("login failed", zap.String("status", resp.Status), zap.String("response", string(respBody)))
+		log.Error("Login failed",
+			zap.String("status", resp.Status),
+			zap.String("response", string(respBody)))
 		return fmt.Errorf("login failed: %s", resp.Status)
 	}
 
 	// Retrieve CSRF token from the response headers
 	if csrf := resp.Header.Get("x-csrf-token"); csrf != "" {
 		c.csrf = resp.Header.Get("x-csrf-token")
 	}
-
 	return nil
 }
 
-// doRequest makes an HTTP request to the UniFi controller.
 func (c *httpClient) doRequest(method, path string, body io.Reader) (*http.Response, error) {
-	log.Debug(fmt.Sprintf("making %s request to %s", method, path))
+	log.Debug("Making request", zap.String("method", method), zap.String("path", path))
+
+	// Convert body to bytes for logging and reuse
+	var bodyBytes []byte
```

Comment:
Little worried about reading the body here, I tried doing it last time and it messed up exdns

Reply:
I am not very comfortable with GO so if you think this can cause trouble, please remove it.

Reply:
> I am not very comfortable with GO so if you think this can cause trouble, please remove it.

It's fine, we will test and verify it functions.